src/selection.ts

```diff
@@ -1,5 +1,33 @@
 import dom from './dom';
 
+const isIframe = (node: Node) => node.nodeName === 'IFRAME';
+const isImg = (node: Node) => node.nodeName === 'IMG';
+
+export const cleanSelection = (selection: Selection): Selection => {
+  const anchor = selection.anchorNode;
+  const focus = selection.focusNode;
+
+  if (!anchor || !focus) {
+    return selection;
+  }
+
+  const anchorParent = anchor.parentNode;
+  const focusParent = focus.parentNode;
+  const beginsOnIframe = isIframe(anchor);
+  const endsOnIframe = isIframe(focus);
+
+  if (beginsOnIframe && anchorParent && !endsOnIframe) {
+    selection.setBaseAndExtent(anchorParent, selection.anchorOffset, focus, selection.focusOffset);
+  }
```

Comment:
Added to cover highlights starting with an image where the image is the first element on the page.

Reply:
please put some comments in the code for the purpose of each manipulation